packages/panzer/disc-fe/src/Panzer_PointValues2.hpp

```diff
@@ -44,82 +44,124 @@
+    operator=(const PointValues2<SourceScalar>& source);
+
+    PointValues2(const std::string & pre="",
+                 bool allocArrays=false)
+       : alloc_arrays_(allocArrays), prefix_(pre) {}
+
+    PointValues2(const std::string & pre,
+                 const std::vector<PHX::index_size_type> & ddims,
+                 bool allocArrays=false)
+       : alloc_arrays_(allocArrays), prefix_(pre), ddims_(ddims) {}
     
     //! Sizes/allocates memory for arrays
-    template <typename ArrayFactory>
-    void setupArrays(const Teuchos::RCP<const panzer::PointRule>& pr,const ArrayFactory & af);
+    void setupArrays(const Teuchos::RCP<const panzer::PointRule>& pr);
+
+    /** Evaluate teh jacobian and derivative information at the requested reference
+      * points.
+      *
+      * \param[in] node_coords Cell vertices
+      * \param[in] point_coords Reference cell coordinates
+      */
+    template <typename CoordinateArray,typename PointArray>
+    void evaluateValues(const CoordinateArray & node_coords,
+                        const PointArray & point_coords)
+    { copyNodeCoords(node_coords);
+      copyPointCoords(point_coords);
+      evaluateValues(); }
+
+    /** Evaluate teh jacobian and derivative information at the requested reference
+      * points. This version allows a shallow copy of the vertex coordinates. 
+      *
+      * \param[in] node_coords Cell vertices
+      * \param[in] point_coords Reference cell coordinates
+      * \param[in] shallow_copy_nodes Enable or disable a shallow copy of the vertices
+      */ 
+    template <typename PointArray>
+    void evaluateValues(const PHX::MDField<Scalar,Cell,NODE,Dim> & node_coords,
+                        const PointArray & point_coords, 
+                        bool shallow_copy_nodes)
+    { if(shallow_copy_nodes)
+        node_coordinates = node_coords;
+      else
+        copyNodeCoords(node_coords);
+      copyPointCoords(point_coords);
+      evaluateValues(); }
+
+    //! Return reference cell coordinates this class uses (IP,Dim) sized
+    PHX::MDField<Scalar,IP,Dim> & getRefCoordinates() const 
+    { return coords_ref; }
+
+    //! Return the vertex coordinates this class uses (Cell,NODE,Dim) sized
+    PHX::MDField<Scalar,Cell,NODE,Dim> & getVertexCoordinates() const
+    { return node_coordinates; }
+
+    // input fields
+    mutable PHX::MDField<Scalar,IP,Dim,void,void,void,void,void,void> coords_ref;      // <IP,Dim>
+    mutable PHX::MDField<Scalar,Cell,NODE,Dim,void,void,void,void,void> node_coordinates; // <Cell,NODE,Dim>
+
+    // output fields
+    PHX::MDField<Scalar,Cell,IP,Dim,Dim,void,void,void,void> jac;              // <Cell,IP,Dim,Dim>
+    PHX::MDField<Scalar,Cell,IP,Dim,Dim,void,void,void,void> jac_inv;          // <Cell,IP,Dim,Dim>
+    PHX::MDField<Scalar,Cell,IP,void,void,void,void,void,void> jac_det;        // <Cell,IP>
+    PHX::MDField<Scalar,Cell,IP,Dim,void,void,void,void,void> point_coords;    // <Cell,IP,Dim> // cell points
```

Comment:
Why are these fields explicitly declaring the "void" parameters here? They should default to void if you leave them out.